beep/dataset.py

```diff
@@ -400,39 +408,45 @@ def get_threshold_targets(dataset_diagnostic_properties,
         elif run_target_df['fractional_metric'].min() > threshold and extrapolate_threshold:
             fill_value = "extrapolate"
             bounds_error = False
-            x_throughput_linspace = np.linspace(x_throughput_axis.min(), 2 * x_throughput_axis.max(), num=1000)
-            x_cycle_linspace = np.linspace(x_cycle_axis.min(), 2 * x_cycle_axis.max(), num=1000)
+            x_linspaces = []
+            for axis in x_axes:
+                x_linspaces.append(np.linspace(axis.min(), 2 * axis.max(), num=1000))
         else:
             fill_value = np.nan
             bounds_error = True
-            x_throughput_linspace = np.linspace(x_throughput_axis.min(), x_throughput_axis.max(), num=1000)
-            x_cycle_linspace = np.linspace(x_cycle_axis.min(), x_cycle_axis.max(), num=1000)
+            x_linspaces = []
+            for axis in x_axes:
+                x_linspaces.append(np.linspace(axis.min(), axis.max(), num=1000))
 
-        f_throughput = interpolate.interp1d(x_throughput_axis, y_interpolation_axis, kind='linear',
-                                            bounds_error=bounds_error, fill_value=fill_value)
-        f_cycle = interpolate.interp1d(x_cycle_axis, y_interpolation_axis, kind='linear', bounds_error=bounds_error,
-                                       fill_value=fill_value)
+        f_axis = []
```

Comment:
Can we move the this initialization of f_axis to below that of x_linspaces, then combine the below code (second for loop over x_axes) to with the first for loop over x_axes?